src/coreclr/interpreter/compiler.cpp

```diff
@@ -2506,17 +2506,13 @@ void InterpCompiler::EmitCall(CORINFO_RESOLVED_TOKEN* constrainedClass, bool rea
                 }
                 m_pLastNewIns->data[0] = GetDataItemIndex(callInfo.hMethod);
             }
-            else if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && !readonly)
+            else if ((callInfo.classFlags & CORINFO_FLG_ARRAY) && newObj)
             {
-                CORINFO_SIG_INFO ctorSignature;
-                CORINFO_CLASS_HANDLE ctorClass;
-
-                m_compHnd->getMethodSig(resolvedCallToken.hMethod, &ctorSignature);
-                ctorClass = m_compHnd->getMethodClass(resolvedCallToken.hMethod);
+                CORINFO_CLASS_HANDLE arrayClsHnd = m_compHnd->getMethodClass(resolvedCallToken.hMethod);
 
                 AddIns(INTOP_NEWMDARR);
-                m_pLastNewIns->data[0] = GetDataItemIndex(ctorClass);
-                m_pLastNewIns->data[1] = numArgs;
+                m_pLastNewIns->data[0] = GetDataItemIndex(arrayClsHnd);
```

Comment:
```suggestion
                m_pLastNewIns->data[0] = GetDataItemIndex(resolvedCallToken.hClass);
```
You should be able to use hClass from resolvedCallToken. No need to call VM to compute it again